apollo-router/tests/snapshots/set_context__set_context_with_null.snap

```diff
@@ -22,14 +22,14 @@ expression: response
               "kind": "Fetch",
               "serviceName": "Subgraph1",
               "variableUsages": [],
-              "operation": "query Query_Null_Param__Subgraph1__0{t{__typename prop id u{__typename id}}}",
+              "operation": "query Query_Null_Param__Subgraph1__0 { t { __typename prop id u { __typename id } } }",
               "operationName": "Query_Null_Param__Subgraph1__0",
               "operationKind": "query",
               "id": null,
               "inputRewrites": null,
               "outputRewrites": null,
               "contextRewrites": null,
-              "schemaAwareHash": "4c0c9f83a57e9a50ff1f6dd601ec0a1588f1485d5cfb1015822af4017263e807",
+              "schemaAwareHash": "4fc423a49bbddcc8869c014934dfd128dd61a1760c4eb619940ad46f614c843b",
```

Comment:
This is computed by `QueryHashVisitor` in `apollo-router/src/spec/query/change.rs`, and includes many details (including directives) from the _subgraph_ schema extracted by the planner from the supergraph schema. The two planners do this extraction differently. Full diff from this specific test:

```diff
-schema @link(url: "https://specs.apollo.dev/link/v1.0") @link(url: "https://specs.apollo.dev/federation/v2.9", import: ["@key", "@requires", "@provides", "@external", "@tag", "@extends", "@shareable", "@inaccessible", "@override", "@composeDirective", "@interfaceObject"]) {
+schema {
   query: Query
 }
 
+extend schema @link(url: "https://specs.apollo.dev/link/v1.0") @link(url: "https://specs.apollo.dev/federation/v2.9")
+
 directive @link(url: String, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA
 
-directive @key(fields: federation__FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE
+directive @federation__key(fields: federation__FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE
 
-directive @requires(fields: federation__FieldSet!) on FIELD_DEFINITION
+directive @federation__requires(fields: federation__FieldSet!) on FIELD_DEFINITION
 
-directive @provides(fields: federation__FieldSet!) on FIELD_DEFINITION
+directive @federation__provides(fields: federation__FieldSet!) on FIELD_DEFINITION
 
-directive @external(reason: String) on OBJECT | FIELD_DEFINITION
+directive @federation__external(reason: String) on OBJECT | FIELD_DEFINITION
 
-directive @tag(name: String!) repeatable on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION | SCHEMA
+directive @federation__tag(name: String!) repeatable on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION | SCHEMA
 
-directive @extends on OBJECT | INTERFACE
+directive @federation__extends on OBJECT | INTERFACE
 
-directive @shareable repeatable on OBJECT | FIELD_DEFINITION
+directive @federation__shareable on OBJECT | FIELD_DEFINITION
 
-directive @inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+directive @federation__inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
 
-directive @override(from: String!, label: String) on FIELD_DEFINITION
+directive @federation__override(from: String!, label: String) on FIELD_DEFINITION
 
-directive @composeDirective(name: String) repeatable on SCHEMA
+directive @federation__composeDirective(name: String) repeatable on SCHEMA
 
-directive @interfaceObject on OBJECT
+directive @federation__interfaceObject on OBJECT
 
 directive @federation__authenticated on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM
 
 directive @federation__requiresScopes(scopes: [[federation__Scope!]!]!) on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM
 
-directive @federation__policy(policies: [[federation__Policy!]!]!) on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM
+directive @federation__cost(weight: Int!) on ARGUMENT_DEFINITION | ENUM | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | OBJECT | SCALAR
 
-directive @federation__sourceAPI repeatable on SCHEMA
+directive @federation__listSize(assumedSize: Int, slicingArguments: [String!], sizedFields: [String!], requireOneSlicingArgument: Boolean = true) on FIELD_DEFINITION
 
-directive @federation__sourceType repeatable on OBJECT | INTERFACE
+directive @federation__fromContext(field: String) on ARGUMENT_DEFINITION
 
-directive @federation__sourceField repeatable on FIELD_DEFINITION
+directive @federation__context(name: String) repeatable on INTERFACE | OBJECT | UNION
 
-directive @federation__context(name: String!) repeatable on INTERFACE | OBJECT | UNION
-
-directive @federation__fromContext(field: federation__ContextFieldValue) on ARGUMENT_DEFINITION
-
-directive @federation__cost(weight: Int!) on ARGUMENT_DEFINITION | ENUM | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | OBJECT | SCALAR
+scalar link__Import
 
-directive @federation__listSize(assumedSize: Int, slicingArguments: [String!], sizedFields: [String!], requireOneSlicingArgument: Boolean = true) on FIELD_DEFINITION
-
 enum link__Purpose {
   """
-  `SECURITY` features provide metadata necessary to securely resolve fields.
+  \`SECURITY\` features provide metadata necessary to securely resolve fields.
   """
   SECURITY
   """
-  `EXECUTION` features provide metadata necessary for operation execution.
+  \`EXECUTION\` features provide metadata necessary for operation execution.
   """
   EXECUTION
 }
 
-scalar link__Import
-
 scalar federation__FieldSet
 
 scalar federation__Scope
 
-scalar federation__Policy
-
-scalar federation__ContextFieldValue
-
 type Query {
   t: T!
   tList: [T]!
@@ -271,34 +261,34 @@
 
 union K @federation__context(name: "context2") = A | B
 
-type A @key(fields: "id") {
+type A @federation__key(fields: "id", resolvable: true) {
   id: ID!
   v: V!
   prop: String!
 }
 
-type B @key(fields: "id") {
+type B @federation__key(fields: "id", resolvable: true) {
   id: ID!
   v: V!
   prop: String!
 }
 
-type T @key(fields: "id") @federation__context(name: "context") {
+type T @federation__key(fields: "id", resolvable: true) @federation__context(name: "context") {
   id: ID!
   u: U!
   uList: [U]!
   prop: String!
 }
 
-type U @key(fields: "id") {
-  id: ID! @shareable
+type U @federation__key(fields: "id", resolvable: true) {
+  id: ID! @federation__shareable
   field(
     a: String @federation__fromContext(field: "$context { prop }"),
   ): Int!
 }
 
-type V @key(fields: "id") {
-  id: ID! @shareable
+type V @federation__key(fields: "id", resolvable: true) {
+  id: ID! @federation__shareable
   field(
     a: String @federation__fromContext(field: "$context2 ... on A { prop } ... on B { prop }"),
   ): Int!